### PR TITLE
Fix license and url of libcurl in conanfile.py

### LINKED ISSUE #12566: [package] libcurl/7.10.4+: Wrong license (MIT)
### Package and Environment Details

* Package Name/Version: **libcurl/7.10.4+**
* Operating System+version: (not relevant)
* Compiler+version:  (not relevant)
* Docker image:  (not relevant)
* Conan version:  (not relevant)
* Python version:  (not relevant)

### Conan profile

_No response_

### Steps to reproduce

The libcurl is published under its own license (SPDX-Identifier: "curl") - "inspired" by MIT but not identical - since version 7.10.4.
See
 - https://curl.se/docs/copyright.html
 - https://spdx.org/licenses/curl.html
 - compare https://github.com/curl/curl/tree/curl-7_10_4 to https://github.com/curl/curl/tree/curl-7_10_3

This should be fixed in https://github.com/conan-io/conan-center-index/blob/master/recipes/libcurl/all/conanfile.py

### Logs

_No response_

### LINKED ISSUE #12567: [package] libcurl/7.74.0+: Outdated URL (https://curl.haxx.se)
### Package and Environment Details

* Package Name/Version: **libcurl/7.74.0+**
* Operating System+version: (not relevant)
* Compiler+version: (not relevant)
* Docker image: (not relevant)
* Conan version: (not relevant)
* Python version: (not relevant)

### Conan profile

_No response_

### Steps to reproduce

The official homepage of libcurl is https//curl.se since version 7.74.0 (released 2020-12-09).
See:
 - https://curl.se/changes.html#7_74_0
 - https://github.com/curl/curl/pull/6172
(Old homepage https://curl.haxx.se forwards to curl.se)

Should be fixed in https://github.com/conan-io/conan-center-index/blob/master/recipes/libcurl/all/conanfile.py

### Logs

<details><summary>Click to expand log</summary>

```
Put your log output here
```

</details>

### PR DESCRIPTION
**Pull request by BjoernAtBosch**
libcurl is provided under "curl" license - see https://curl.se/docs/copyright.html and https://spdx.org/licenses/curl.html
URL is https://curl.se

Specify library name and version:  **libcurl/7.10.4+**

fixes #12566 
fixes #12567

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -13,9 +13,9 @@
 class LibcurlConan(ConanFile):
     name = "libcurl"
     description = "command line tool and library for transferring data with URLs"
-    license = "MIT"
+    license = "curl"
```
Comment (uilianries) [inline recipes/libcurl/all/conanfile.py:16]: Also, you need to update:

```python
def config_options(self):
    if Version(self.version) < "7.10.4":
        self.license = "MIT"
```

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution. Please, don't forget to sign the CLA and ask for access on the issue #4.

File: recipes/libcurl/all/conanfile.py (hunk 1 above)
Comment (BjoernAtBosch) [reply recipes/libcurl/all/conanfile.py:16]: Done :-)

**Review by prince-chrismc — DISMISSED** [review]

**Comment by uilianries** [comment]
@BjoernAtBosch Please, consider https://github.com/BjoernAtBosch/conan-center-index/pull/1. It should fix your linter errors.

**Review by uilianries — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Pull request merged**